packages/statemanager/tests/ethersStateManager.spec.ts

```diff
@@ -0,0 +1,112 @@
+
+  t.ok(
+    typeof (state as any).contractCache.get(UNIerc20ContractAddress.toString()) !== 'undefined',
+    'UNI ERC20 contract code was found in cache'
+  )
+  const storageSlot = await state.getContractStorage(
+    UNIerc20ContractAddress,
+    setLengthLeft(bigIntToBuffer(1n), 32)
+  )
+  t.ok(storageSlot.length > 0, 'was able to retrieve storage slot 1 for the UNI contract')
+  t.notEqual(
+    (state as any).storageCache.get(
+      UNIerc20ContractAddress + '--' + bufferToHex(setLengthLeft(bigIntToBuffer(1n), 32))
+    ),
+    undefined,
+    'a storage slot for the UNI contract exists in the cache'
+  )
+
+  const stateRoot = await state.getStateRoot()
+  t.ok(
+    stateRoot !== undefined && (state as any).root === undefined,
+    'state root was pulled from provider and not cached when blockTag is "latest"'
+  )
+  t.end()
+})
+
+tape('runTx tests', async (t) => {
+  const common = new Common({ chain: Chain.Mainnet })
+  const state = new EthersStateManager({ provider: provider })
+  const vm = await VM.create({ common, stateManager: state })
+
+  const vitalikDotEth = Address.fromString('0xd8da6bf26964af9d7eed9e03e53415d37aa96045')
+  const privateKey = Buffer.from(
+    'e331b6d69882b4cb4ea581d88e0b604039a3de5967688d3dcffdd2270c0fd109',
+    'hex'
+  )
+  const tx = FeeMarketEIP1559Transaction.fromTxData(
+    { to: vitalikDotEth, value: '0x100', gasLimit: 500000n, maxFeePerGas: 7 },
+    { common }
+  ).sign(privateKey)
+  const result = await vm.runTx({
+    skipBalance: true,
+    skipNonce: true,
+    tx,
+  })
+
+  t.equal(result.totalGasSpent, 21000n, 'sent some ETH to vitalik.eth')
+  t.end()
+})
+
+/** To run the block test, you will need an Infura or Alchemy URL that gives access to complete chain history.
+ *  Pass it in the PROVIDER=[provider url] npm run tape -- 'tests/ethersStateManager.spec.ts'
+ *  Cloudflare only provides access to the last 128 blocks so throws errors on this test.
+ */
+
+tape.only('runBlock test', async (t) => {
+  if (process.env.PROVIDER === undefined) t.fail('no provider URL provided')
+  const common = new Common({ chain: Chain.Mainnet, hardfork: Hardfork.Chainstart })
+  const provider = new JsonRpcProvider(process.env.PROVIDER)
+  const blockTag = BigInt(
+    (await provider.send('eth_getBlockByNumber', [bigIntToHex(71980n), false])).number
+  )
+  const state = new EthersStateManager({
+    provider: provider,
+    // Set the state manager to look at the state of the chain before the block has been executed
+    blockTag: blockTag - 1n,
+  })
+  const vm = await VM.create({ common, stateManager: state })
+  const blockData = await provider.send('eth_getBlockByNumber', [bigIntToHex(blockTag), true])
+  const block = blockFromRpc(blockData, undefined, { common, hardforkByBlockNumber: true})
+  const res = await vm.runBlock({
+    block,
+    skipHeaderValidation: true,
+    skipBlockValidation: true,
+    skipBalance: true,
+    skipNonce: true,
```

Comment:
Just for my understanding, `skipBalance` and `skipNonce` is still needed in this get-the-state-from-a-central-source scenario?

Reply:
I removed those from the test and you actually don't need them in this scenario.  I think you would need those if you were constructing a custom transaction and didn't want to deal with balance and nonce checks on the caller account but they pass just fine if you set up the preconditions correctly.  In the case of just trying to run a block from mainnet, the preconditions you need are to set up the state manager to make it's calls against the state at the [block number - 1](https://github.com/ethereumjs/ethereumjs-monorepo/blob/0b7f966ff570160b40518a2b9db0a86297c45a17/packages/statemanager/tests/ethersStateManager.spec.ts#L98) and then retrieve the block and try and run it.  if the state manager is pulling valid state from the provider, everything should pass and the block state root you get from the VM should match what's in the block pulled from the provider.  If you point the state manager to the wrong block height, you may get nonce validation failures (for example) if you run a block with a transaction from a calling account in it and the state manager is looking pulling state from the chain at some other height.
In this precise case, you still need `skipHeaderValidation` because the parent block isn't in the vm's blockchain so it returns header not found.  We could make a separate blockchain implementation if we wanted to pull the headers from the provider as well.

There's still some nuance to conditioning the tests, but really just setting up which validation rules you want to use.